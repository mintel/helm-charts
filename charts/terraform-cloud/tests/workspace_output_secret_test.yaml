suite: Test Workspace Output Secret
templates:
  - workspace-output-secret.yaml
release:
  namespace: test-namespace
tests:
  - it: Dev S3 module instance workspace output secret with override
    set:
      global.name: test-app
      global.clusterEnv: dev
      global.clusterName: cluster1
      global.clusterRegion: eu-west-1
      global.owner: sre
      global.partOf: test-project
      global.terraform.externalSecret: true
      s3:
        enabled: true
        terraform:
          instances:
            test-bucket:
              outputSecretMap:
                TEST_BUCKET_ID: BUCKET_ID
    asserts:
      - matchSnapshot: {} # Check for regressions and unexpected changes.
      - isKind:
          of: ExternalSecret
      - equal:
          path: metadata.namespace
          value: test-namespace
      - equal:
          path: metadata.name
          value: mntl-test-bucket-s3
      - contains:
          path: spec.data
          content:
            secretKey: TEST_BUCKET_ID
            remoteRef:
              key: test-namespace/mntl-test-bucket/s3
              property: BUCKET_ID
      - isNull:
          path: spec.dataFrom
      - equal:
          path: spec.secretStoreRef.kind
          value: SecretStore
      - hasDocuments:
          count: 1
      - equal:
          path: spec.secretStoreRef.name
          value: aws-secrets-manager-local
  - it: Dev S3 module workspace output secret with override
    set:
      global.name: test-app
      global.clusterEnv: dev
      global.clusterName: cluster1
      global.clusterRegion: eu-west-1
      global.owner: sre
      global.partOf: test-project
      global.terraform.externalSecret: true
      s3:
        enabled: true
        terraform:
          defaultVars:
            outputSecretMap:
              TEST_BUCKET_ID: BUCKET_ID
    asserts:
      - matchSnapshot: {} # Check for regressions and unexpected changes.
      - isKind:
          of: ExternalSecret
      - equal:
          path: metadata.namespace
          value: test-namespace
      - equal:
          path: metadata.name
          value: mntl-test-app-s3
      - contains:
          path: spec.data
          content:
            secretKey: TEST_BUCKET_ID
            remoteRef:
              key: test-namespace/mntl-test-app/s3
              property: BUCKET_ID
      - isNull:
          path: spec.dataFrom
      - equal:
          path: spec.secretStoreRef.kind
          value: SecretStore
      - hasDocuments:
          count: 1
      - equal:
          path: spec.secretStoreRef.name
          value: aws-secrets-manager-local
  - it: Dev S3 module workspace output secret
    set:
      global.name: test-app
      global.clusterEnv: dev
      global.clusterName: cluster1
      global.clusterRegion: eu-west-1
      global.owner: sre
      global.partOf: test-project
      global.terraform.externalSecret: true
      s3.enabled: true
    asserts:
      - matchSnapshot: {} # Check for regressions and unexpected changes.
      - isKind:
          of: ExternalSecret
      - equal:
          path: metadata.namespace
          value: test-namespace
      - equal:
          path: metadata.name
          value: mntl-test-app-s3
      - equal:
          path: spec.dataFrom[0].key
          value: test-namespace/mntl-test-app/s3
      - equal:
          path: spec.secretStoreRef.kind
          value: SecretStore
      - hasDocuments:
          count: 1
      - equal:
          path: spec.secretStoreRef.name
          value: aws-secrets-manager-local
