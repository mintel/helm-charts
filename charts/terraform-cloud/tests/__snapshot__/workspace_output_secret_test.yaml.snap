Dev S3 module instance workspace output secret with override:
  1: |
    apiVersion: external-secrets.io/v1beta1
    kind: ExternalSecret
    metadata:
      annotations:
        app.mintel.com/placeholder: placeholder
        argocd.argoproj.io/sync-options: SkipDryRunOnMissingResource=true
        argocd.argoproj.io/sync-wave: "-20"
      labels:
        app.mintel.com/env: dev
        app.mintel.com/owner: sre
        app.mintel.com/region: eu-west-1
        name: mntl-test-bucket-s3
      name: mntl-test-bucket-s3
      namespace: test-namespace
    spec:
      data:
        - remoteRef:
            key: test-namespace/mntl-test-bucket/s3
            property: BUCKET_ID
          secretKey: TEST_BUCKET_ID
      refreshInterval: 5m0s
      secretStoreRef:
        kind: SecretStore
        name: aws-secrets-manager-local
      target:
        creationPolicy: Owner
Dev S3 module workspace output secret:
  1: |
    apiVersion: external-secrets.io/v1beta1
    kind: ExternalSecret
    metadata:
      annotations:
        app.mintel.com/placeholder: placeholder
        argocd.argoproj.io/sync-options: SkipDryRunOnMissingResource=true
        argocd.argoproj.io/sync-wave: "-20"
      labels:
        app.mintel.com/env: dev
        app.mintel.com/owner: sre
        app.mintel.com/region: eu-west-1
        name: mntl-test-app-s3
      name: mntl-test-app-s3
      namespace: test-namespace
    spec:
      dataFrom:
        - extract:
            key: test-namespace/mntl-test-app/s3
      refreshInterval: 5m0s
      secretStoreRef:
        kind: SecretStore
        name: aws-secrets-manager-local
      target:
        creationPolicy: Owner
Dev S3 module workspace output secret with override:
  1: |
    apiVersion: external-secrets.io/v1beta1
    kind: ExternalSecret
    metadata:
      annotations:
        app.mintel.com/placeholder: placeholder
        argocd.argoproj.io/sync-options: SkipDryRunOnMissingResource=true
        argocd.argoproj.io/sync-wave: "-20"
      labels:
        app.mintel.com/env: dev
        app.mintel.com/owner: sre
        app.mintel.com/region: eu-west-1
        name: mntl-test-app-s3
      name: mntl-test-app-s3
      namespace: test-namespace
    spec:
      data:
        - remoteRef:
            key: test-namespace/mntl-test-app/s3
            property: BUCKET_ID
          secretKey: TEST_BUCKET_ID
      refreshInterval: 5m0s
      secretStoreRef:
        kind: SecretStore
        name: aws-secrets-manager-local
      target:
        creationPolicy: Owner
